sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -413,6 +415,49 @@ object IntervalUtils {
     new CalendarInterval(truncatedMonths, truncatedDays, micros.round)
   }
 
+  /**
+   * Makes an interval from months, days and micros with the fractional part by
+   * adding the month fraction to days and the days fraction to micros.
+   */
+  private def safeFromDoubles(
+      monthsWithFraction: Double,
+      daysWithFraction: Double,
+      microsWithFraction: Double): CalendarInterval = {
+    val monthInLong = monthsWithFraction.toLong
+    val truncatedMonths = if (monthInLong > Int.MaxValue) {
```

Comment:
It's not truncation that only drops the fraction part. I think it's hard to define "safe" for this `fromDoubles` method. Maybe we don't need a safe version for interval multiple (and we never have)

how about
```
def multiplyExact(interval: CalendarInterval, num: Double): CalendarInterval = {
  fromDoubles(num * interval.months, num * interval.days, num * interval.microseconds)
}

def multiple(interval: CalendarInterval, num: Double): CalendarInterval = multiplyExact(interval, num)
```

Reply:
hmm, then for multiply/divide overflow we return `null`s for other arithmetics we return overflowed values? this seems weird too

Reply:
the behavior with current implementation follows `select cast(9223372036854775807L * 1.1D as Long);` results 9223372036854775807L

Reply:
We add the overflow flag so that we can avoid breaking backward compatibility. We don't have this concern for interval multiple.

For the overflow behavior, I think it's caused by `Double.toLong`. Is there an exact version of it? if not we may need to check `double > Long.Max` first.

Reply:
`WHEN ANSI OFF`
There seems no 
1. If we use `NULL ON OVERFLOW` policy for all arithmetics, means interval `add/subtract/negate` break their backward compatibility to fit `multiply/divide`, seem not a good choice.
2. If we remain to use `NULL ON OVERFLOW` policy for `multiply/divide` and java overflow style for `add `etc, this breaks no backward compatibility but also keeps the inconsistency. seem not a good choice either.
3. If we throw exceptions for `multiply/divide` for overflows and remain java overflow style for `add` etc, this breaks no backward compatibility but also raises much worse inconsistency issue.
4. If we throw exceptions for all interval arithmetics, this is just `ANSI ON`.
5. Behavior of current implementation, we do not break backward compatibility either.  Things only go wrong when it reaches the `CalendarInterval.MAX_VALUE`. It seems the best choice among the worst.

Things became odd since we didn't support fraction representations for intervals, but support `multiply/divide` intervals with fractions. a.k.a we do not have consistent rules https://github.com/apache/spark/pull/26592 for how the fraction part of month go days, how day fraction go milliseconds.

There seems no other systems can refer to for our `ANSI OFF` mode.

Reply:
how about always fail interval multiple when overflow? ANSI off doesn't mean everything is against ANSI standard, it means to keep the existing behavior even if it violates ANSI standard.

Reply:
ok, make sense